File: string/palindrome-string.py
class Solution:
    def isPalindrome(self, A):
        low = 0
        high = len(A) - 1

        while low < high:
            if not A[low].isalnum(): low += 1
            elif not A[high].isalnum(): high -= 1
            else:
                if A[low].lower() != A[high].lower(): return 0
                low += 1
                high -= 1

        return 1

File: string/palindrome-string.cpp
bool valid(char a){
    a = tolower(a);
    return (a>='0' && a<='9') || (a>='a' && a<='z');
}
 
int Solution::isPalindrome(string A) {
    int n = A.size();
    int i = 0, j = n - 1, k = 0;

    while(i < j){
        while(!valid(A[i])) i++;
        while(!valid(A[j])) j--;
        if(tolower(A[i]) != tolower(A[j])) return 0;
        i++; j--;
    }
    return 1;
}

